azavea/pfb-network-connectivity
# Feature/analysis job endpoint

### Add API endpoints for Neighborhood an AnalysisJob

Run `AnalysisJob` on creation, or if updated to reset its status to `CREATED`.

### Testing

 - Clear existing database with `./scripts/django-manage reset_db`
 - Run migrations: `./scripts/django-manage migrate`
 - Ensure AWS `pfb` configuration exists, as per README
 - Create a neighborhood using the form at http://localhost:9200/api/neighborhoods/
 - Create an analysis job using the form at http://localhost:9200/api/analysis_jobs/

Neighborhood and analysis job should be created successfully. Creation of analysis job should kick of job run, set batch job ID, and update status.

### Notes

After this is merged, will need to clear database, or at least delete existing neighborhoods and/or the default user, then re-run migrations. This is to support addition of auditing fields to neighborhoods.

Updated max upload size nginx allows to 6MB from the default 1MB, somewhat arbitrarily; not sure what the largest shapefile zip upload would be that we should support.

Connects #127 and connects #93.
